backend/src/types.ts

```diff
@@ -148,6 +148,18 @@ export type K8sResourceCommon = {
   };
 } & K8sResourceBase;
 
+export type K8sResourceCommonWithMetadataAndNamespace = {
+  metadata: {
+    name?: string;
+    namespace: string;
+    generateName?: string;
+    uid?: string;
+    labels?: { [key: string]: string };
+    annotations?: { [key: string]: string };
+    creationTimestamp?: Date;
+  };
+} & K8sResourceBase;
```

Comment:
Does this work? (also made the type name smaller)
```suggestion
export type K8sNamespacedResourceCommon = {
  metadata: {
    namespace: string;
  };
} & K8sResourceCommon;
```